### ISSUE TITLE
Tuned Bert Model on MRPC gives wrong predictions.

### ISSUE THREAD

**Issue by hima950** [google-research/bert#663]
Hi There,
I have tried to implement the example shown on sentence pair classification using MRPC data-set. I have trained the model with one epoch for testing purpose and I get a 0.68% accuracy.
But, the problem is when predicting the labels for the "test.tsv" it predicts too high probabilities all close to one for all samples.
**Here is my Training script:**

```
export BERT_BASE_DIR=/home/herla/notebooks/text-analysis/uncased_L-12_H-768_A-12
export TRAINED_CLASSIFIER=/home/herla/notebooks/text-analysis/uncased_L-12_H-768_A-12/bert_model.ckpt
```

`python3 run_classifier.py --task_name=MRPC --do_train=true --do_eval=true --data_dir=./data --use_gpu=True --vocab_file=$BERT_BASE_DIR/vocab.txt --bert_config_file=$BERT_BASE_DIR/bert_config.json --init_checkpoint=$BERT_BASE_DIR/bert_model.ckpt --max_seq_length=128 --train_batch_size=32 --learning_rate=2e-5 --num_train_epochs=1.0 --output_dir=./bert_output/
`

**Here is the Prediction Script:**

```
export BERT_BASE_DIR=/home/herla/notebooks/text-analysis/uncased_L-12_H-768_A-12
export TRAINED_CLASSIFIER=./bert_output/model.ckpt-1520
```
```
python3 run_classifier.py \
--task_name=MRPC \
--use_gpu=True \
--do_predict=true \
--data_dir=./data \
--vocab_file=$BERT_BASE_DIR/vocab.txt \
--bert_config_file=$BERT_BASE_DIR/bert_config.json \
--init_checkpoint=$TRAINED_CLASSIFIER \
--max_seq_length=128 \
--output_dir=./bert_output/
```

The format of the files is correct as I use the directly downloaded MRPC data-sets. Here is the sample of the predictions:

> 0.0013405858	0.9986594
> 0.0013392371	0.9986608
> 0.0013387766	0.99866116
> 0.0013370833	0.99866295
> 0.0013388416	0.99866116
> 0.0013383598	0.99866164
> 0.001341351	0.99865866
> 0.0013357978	0.99866426
> 0.0013370474	0.99866295
> 0.0013380272	0.998662
> 0.0013391515	0.9986608
> 0.001335816	0.99866414
> 0.0013396177	0.9986603
> 0.0013387485	0.9986613
> 0.0013373283	0.9986627
> 0.00133698	0.99866307
> 0.001340849	0.99865913
> 0.0013396387	0.9986603
> 0.0013360337	0.998664
> 0.0013410482	0.9986589
> 0.0013396874	0.9986603
> 

**All the samples are predicted similarly. I have also tried with my own train-set generated from a wiki corpus still get predictions like this. 
I do not understand what I am doing wrong could you please help me resolve this issue.**
Thanks in advance.
Hima

**Comment by Rotramartya** [comment]
Same problem with me. The predicted probabilities for different labels is same in all the examples of test set

**Comment by dcferreira** [comment]
I might be having the same problem with my dataset, did you figure out the problem by now?

**Comment by Rotramartya** [comment]
@dcferreira The problem with my model was that model was not getting trained properly. I was using just 1 epoch. I changed the hyperparameters (increased epochs, reduced learning rate, etc) and its giving excellent results. Hope this helps

**Comment by wangjs9** [comment]
I have the same problem. It seems that this estimator for prediction is not loaded. But when I tried do_evaluate, the result seems right. I think that there is something wrong with estimator.predict. But I am not sure. Have you figured it out?

**Comment by Parikshit22** [comment]
I'm facing the same issue, run_classifier is predicting the same class every time with same probabilities, i don't know i tried changing the learning rate to bit and also tried increasing the epochs but anything didn't worked well. Also i tried shuffling my data.

**Comment by Maz101** [comment]
You would need to do what Rotramartya says. Increase your epoch training to 100 epochs.

**Comment by Parikshit22** [comment]
Thank you Maz101,
Yes i figured it out it was because of lack of training.
